packages/grafana-ui/src/components/Combobox/Combobox.tsx

```diff
@@ -135,6 +136,24 @@ export const Combobox = <T extends string | number>({
 
   const rowVirtualizer = useVirtualizer(virtualizerOptions);
 
+  const debounceAsync = useMemo(
+    () =>
+      debounce((inputValue: string, customValueOption: ComboboxOption<T> | null) => {
+        loadOptions(inputValue)
+          .then((opts) => {
+            setItems(customValueOption ? [customValueOption, ...opts] : opts);
+            setAsyncLoading(false);
+          })
+          .catch((err) => {
+            if (!(err instanceof StaleResultError)) {
+              // TODO: handle error
+              setAsyncLoading(false);
+            }
+          });
+      }, 300),
```

Comment:
My gut feeling would be to set this pretty low, otherwise there might be a risk of the UI feeling sluggish. (I understand that a bigger value is needed while developing though). Whatever it is, we could probably set a create a const for it.

Do you think it is possible do extract this functionality into a hook? Then we could probably reuse it a bit easier when going Multi

Reply:
300 is what we use for search results, nested folder picker, etc. So I just picked something that is common across Grafana. I can lower it a little bit, but definitely not lower than 200ms, we have that in some places like command palette.

About the hook, I was trying to implement it, but adding a hook with a debounce inside that returns a promise is seeming complicated for the amount we end up gaining on this 😅 I think we can revisit this on Multi if we want, but for now I'd say keep it as is if that's ok.